cpp/models/abm/location.h

```diff
@@ -245,16 +239,30 @@ class Location
         m_capacity_adapted_transmission_risk = consider_capacity;
     }
 
+    /**
+     * Store the current number of individuals in the each infection state at a specific time step
+     * @param t the time step to save
+    */
+    void store_current_population(const TimePoint& t);
+
+    /**
+     * Return the time series object of the current number of individuals in the each infection state
+     * @return the time series object of the current number of individuals in the each infection state
+    */
+    const TimeSeries<double>& get_time_series_subpopulations() const
+    {
+        return m_subpopulations_time_series;
+    }
+
 private:
     void change_subpopulation(InfectionState s, int delta);
-
 private:
     LocationType m_type;
     uint32_t m_index;
     int m_num_persons = 0;
     LocationCapacity m_capacity;
     bool m_capacity_adapted_transmission_risk;
-    std::array<int, size_t(InfectionState::Count)> m_subpopulations;
+    TimeSeries<double> m_subpopulations_time_series;
```

Comment:
@khoanguyen-dev Why did you close it? Did you have some discussion with @DavidKerkmann on that?

Actually I first wanted to go in the same direction as David and even propose to use a TimeSeries<size_t> (if we have implementations for that, not sure). But then I thought it would be easier to use floating point arithmetic as we anyway need this if we want to compute averages of more than one run or interpolate on common time steps.